dashboard/modules/job/job_head.py

```diff
@@ -204,6 +212,80 @@ async def list_jobs(self, req: Request) -> Response:
             content_type="application/json",
         )
 
+    @routes.get("/api/all_jobs/")
+    @optional_utils.init_ray_and_catch_exceptions(connect_to_serve=False)
+    async def list_jobs(self, req: Request) -> Response:
+        driver_jobs, submission_job_drivers = await self._get_driver_jobs()
+        submission_jobs = self._job_manager.list_jobs()
```

Comment:
We need await here too (otherwise, this can block the main thread which is bad for API server).

I believe await is not supported right now due to some tech debt. Can you add TODO here (to use await list_jobs())?

Reply:
should be straightforward, just need to use async internal_kv API I think

Reply:
I'll add a TODO. I don't see an async internal_kv API though.

https://github.com/ray-project/ray/blob/master/python/ray/experimental/internal_kv.py

Reply:
I think you need to use `GcsAioClient`, see https://github.com/ray-project/ray/pull/25788 for an example

Reply:
can we wrap this to thread executor in this case? I'd like to avoid having blocking call in a periodically called API (it's called from dashboard)